src/main/java/org/gaul/s3proxy/S3Proxy.java

```diff
@@ -107,8 +111,12 @@ public final class S3Proxy {
         if (builder.secureEndpoint != null) {
             SslContextFactory.Server sslContextFactory =
                 new SslContextFactory.Server();
-            sslContextFactory.setKeyStorePath(builder.keyStorePath);
-            sslContextFactory.setKeyStorePassword(builder.keyStorePassword);
+            if (builder.sslContext != null) {
+                sslContextFactory.setSslContext(builder.sslContext);
+            } else {
+                sslContextFactory.setKeyStorePath(builder.keyStorePath);
+                sslContextFactory.setKeyStorePassword(builder.keyStorePassword);
+            }
```

Comment:
Do you think it is better to require the caller to create a `SslContextFactory.Server` then pass it to the `Builder`?

Reply:
I also considered that option, but I  thought it would be better to avoid exposing Jetty specifics/internals (like SslContextFactory.Server) to the outside world, and minimize API changes if S3proxy was ever to swap Jetty for something else.

Reply:
Makes sense.